compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -579,6 +581,11 @@ BLangConstantValue constructBLangConstantValueWithExactType(BLangExpression expr
 
         this.anonTypeNameSuffixes = anonTypeNameSuffixes;
         updateConstantType(constantSymbol, expression, env);
+        BLangTypeDefinition typeDefinition = isSourceOnlyAnon ? createdTypeDefinitions.get(constantSymbol.type.tsymbol)
+                : null;
+        if (typeDefinition != null) {
```

Comment:
the reason to have two if conditions (at L584 and L586) is `createdTypeDefinitions.get(constantSymbol.type.tsymbol)` could also be `null`?